### DISCUSSION TITLE
Training based on iterations

### DISCUSSION THREAD

**Discussion by mshooter** [Lightning-AI/pytorch-lightning#12511] [Lightning Trainer API: Trainer, LightningModule, LightningDataModule] [answered]
Hi, 
Could anyone advice me how to set up PyTorch lightning trainer to learn based on iterations instead of epochs? 

Thank you!

**Comment by ananthsub** [comment, marked as answer]
Hi @mshooter , you can use the `min_steps` and `max_steps `arguments on the Trainer to do training based on iterations instead of epochs. https://pytorch-lightning.readthedocs.io/en/latest/common/trainer.html#max-steps

**Reply by mshooter** [reply]
Ok great thanks!
